code-challenges.js

```diff
@@ -2,50 +2,121 @@
+console.log(arrayShuffler(collections))
+console.log(arrayShuffler(collections))
+console.log(arrayShuffler(collections))
+console.log(arrayShuffler(collections))
+console.log(arrayShuffler(collections))
 
 // --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
-var cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-var cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
-
-
-
-
+var cubeAndSum1 = [2, 3, 4] // Expected output: 99
+var cubeAndSum2 = [0, 5, 10] // Expected output: 1125
+
+// Create a function that takes an array of numbers as a argument
+const cubeAndSum = (arr) => {
+	// create a variable to hold the accumulated value
+	let result = 0
+	// loop through array and cube each number and add to accumulating result value
+	arr.map((value) => {
+		return (result = value * value * value + result)
+	})
+	// return the sum of all the numbers cubed
+	return result
+}
+
+// Call the function
+console.log('Problem #2')
+console.log(cubeAndSum(cubeAndSum1))
+console.log(cubeAndSum(cubeAndSum2))
 
 // --------------------3) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
-var nums1 = [3, 56, 90, -8, 0, 23, 6]
-// Expected output: [-8, 90]
-var nums2 = [109, 5, 9, -59, 8, 24]
-// Expected output: [-59, 109]
-
-
-
-
+var nums1 = [3, 56, 90, -8, 0, 23, 6] // Expected output: [-8, 90]
+var nums2 = [109, 5, 9, -59, 8, 24] // Expected output: [-59, 109]
+
+// Create a function that takes an array of numbers as a argument
+const minMax = (arr) => {
+	let resultArray = []
+	// create variable to hold min value
+	let min = arr.reduce((a, b) => {
+		return Math.min(a, b)
+	})
+	// create variable to hold max value
+	let max = arr.reduce((a, b) => {
+		return Math.max(a, b)
+	})
+	// push min and max value to the result array
+	resultArray.push(min, max)
+	// return an array with the minimum and the maximum numbers in that order
+	return resultArray
+}
+
+console.log('Problem #3')
+console.log(minMax(nums1))
+console.log(minMax(nums2))
 
 // --------------------4) Create a function that takes in a string and returns a string with every other letter capitalized.
 
-var testString1 = "albatross"
-// Expected output: "aLbAtRoSs"
-var testString2 = "jabberwocky"
-// Expected output: "jAbBeRwOcKy"
-
-
+var testString1 = 'albatross' // Expected output: "aLbAtRoSs"
+var testString2 = 'jabberwocky' // Expected output: "jAbBeRwOcKy"
+
+// Create a function that takes a string as an argument
+const wonkyLetterMaker = (str) => {
+	return (
+		str
+			// split the string into an array
+			.split('')
+			// loop over the letters in the array and determine odd index and put to correct case
+			.map((letter, index) =>
+				index % 2 === 0 ? letter.toLowerCase() : letter.toUpperCase()
+			)
+			// return array of converted letters as a string with every other letter capitalized
+			.join('')
+	)
+}
```

Comment:
nice work chaining these methods. I might just go in and adjust the indentation? otherwise looks good 👍🏻